mrp_production_hierarchy/models/mrp_production.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ABF OSIELL <http://osiell.com>
+# Copyright 2019 Sergio Corato <https://github.com/sergiocorato>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    root_id = fields.Many2one(
+        'mrp.production', u"Root order",
+        index=True, ondelete='restrict', readonly=True)
+    parent_id = fields.Many2one(
+        'mrp.production', u"Parent order",
+        index=True, ondelete='restrict', readonly=True)
+    child_ids = fields.One2many(
+        'mrp.production', 'parent_id', u"Child orders",
+        domain=[('state', '!=', 'cancel')])
+
+    @api.multi
+    def _generate_moves(self):
+        """Overloaded to pass the created production order ID in the context.
+        It will be used by the 'procurement_order.make_mo()' overload to
+        set the parent relation between production orders.
+        """
+        for prod in self:
+            # Set the initial root production order ID
+            if not prod.env.context.get('root_mrp_production_id'):
+                prod = prod.with_context(root_mrp_production_id=self.id)
+            # Set the parent production order ID
+            prod = prod.with_context(parent_mrp_production_id=self.id)
+            super(MrpProduction, prod)._generate_moves()
+        return True
+
+    @api.multi
+    def open_production_tree(self):
+        self.ensure_one()
+        if self.child_ids:
+            return {
+                'domain': [('root_id', '=', self.id)],
```

Comment:
I didn't test this locally, but this changes the current behavior of the button, which was to be able to open the hierarchy starting from the current MO. Here we are now opening all MOs having the current one as root. Is it intended?

Reply:
This change is needed because `field name="field_parent"` do not exist anymore, or is there an alternative?

Reply:
I don't know unfortunately...
My point was that if we open the tree view from a MO in the middle of the hierarchy, I suspect that nothing will appear (as self.id is not the root MO id)?

Reply:
If there is no hierarchy tree view anymore, we could simply display child MOs with a  `[('id', 'parent_of', self.id)]`. What do you think?

Reply:
In this case, `root_id` field has sense anymore?

Reply:
Probably `[('id', 'child_of', self.id)]`

Reply:
`root_id` store the root MO id, so every MO of a hierarchy share the same root MO (excepting the root MO IIRC, where `root_id` is False). That way it is easy to get the MO from anywhere in the hierarchy (could be used in custom modules).

This field was also used to open easy a hierarchical tree view of any MO (thanks to `field_parent`), but now that this kind of view doesn't exist anymore we could simply open the current MO with its children, without hierarchy.

Reply:
I'm back on this project after a while ;) , `[('id', 'child_of', self.id)]` won't be an improvement as it shows an undefined value corresponding to self:
![Screenshot(66)](https://user-images.githubusercontent.com/7657311/106312316-be127900-6266-11eb-8acd-20c5f953943b.png)
![Screenshot(65)](https://user-images.githubusercontent.com/7657311/106312324-bf43a600-6266-11eb-9257-671c404c998f.png)